scripts/populate_database.py

```
@@ -239,38 +267,48 @@ async def _try_steam_library_fallback(self, app_id: str) -> Optional[dict]:
-            except (ValueError, TypeError, KeyError, OSError) as e:
-                logger.debug(
-                    "Steam library app list lookup failed for %s: %s", app_id, e
-                )
-            return None
+        # Capture steam_api in a local variable to satisfy type checker
+        steam_api = self.steam_api
+
+        async with self._app_list_lock:
+            if self._app_list_cache is None:
+
+                def get_app_list():
```

Ruff (0.15.7)

[warning] Missing return type annotation for private function `get_app_list`

(ANN202)